### PR TITLE
Fix bug with float enums

### PR DESCRIPTION
**Pull request by matthchr**
Previously we would generate code that didn't compile for this case (can't cast string to float).

### CODE REVIEW

File: v2/tools/generator/internal/jsonast/jsonast.go [hunk 1]
```diff
@@ -451,7 +451,7 @@ func enumHandler(ctx context.Context, scanner *SchemaScanner, schema Schema) (as
 		vTrimmed := strings.Trim(v, "\"")
 
 		// Some specs include boolean enums with quotes around the boolean literals
-		if baseType == astmodel.BoolType {
+		if baseType == astmodel.BoolType || baseType == astmodel.FloatType {
```
Comment (theunrepentantgeek) [inline v2/tools/generator/internal/jsonast/jsonast.go:454]: Suggest always trimming quotes _unless_ the baseType is string (e.g. int enums would also need trimming). 
Also need to update the comment.

**Review by theunrepentantgeek — APPROVED** [review]

File: v2/tools/generator/internal/jsonast/jsonast.go (hunk 1 above)
Comment (matthchr) [reply v2/tools/generator/internal/jsonast/jsonast.go:454]: Oops, I thought about the comment but forgot to fix it. Will do as you say.

Comment (matthchr) [reply v2/tools/generator/internal/jsonast/jsonast.go]: Fixed!

**Pull request merged**